### PR TITLE
[aggregation] Ignore metric samples with NaN values

### PR DESCRIPTION
**Pull request by olivielpeau**
### What does this PR do?

Handle `NaN` values the same way we already handle `+Inf`/`-Inf`, i.e. ignore them.

### Motivation

The individual metric aggregators (`Gauge`, `Count`, `Rate`, etc) don't check for this value (and can't do anything with it anyway), and, more importantly (but incidentally), the JSON serialization of the final `Series` downstream fail if there are such values.

### CODE REVIEW

**Review by masci — DISMISSED** [review]
LGTM!

File: pkg/metrics/context_metrics.go [hunk 1]
```diff
@@ -26,8 +26,8 @@ func MakeContextMetrics() ContextMetrics {
 // AddSample add a sample to the current ContextMetrics and initialize a new metrics if needed.
 // TODO: Pass a reference to *MetricSample instead
 func (m ContextMetrics) AddSample(contextKey ckey.ContextKey, sample *MetricSample, timestamp float64, interval int64) {
-	if math.IsInf(sample.Value, 0) {
-		log.Warn("Ignoring sample with +/-Inf value on context key:", contextKey)
+	if math.IsInf(sample.Value, 0) || math.IsNaN(sample.Value) {
+		log.Warn("Ignoring sample with ", sample.Value, " value on context key:", contextKey)
```
Comment (truthbk) [inline pkg/metrics/context_metrics.go]: I fear certain metrics might be consistently in weird ranges fairly consistently, so I would maybe drop the log level?

Comment (olivielpeau) [reply pkg/metrics/context_metrics.go]: yeah that's true, not sure it'd bring much value to spam the logs with `Warn` if we start getting a lot of invalid values. I'll lower this to `Debug`

**Review by truthbk — APPROVED** [review]
👌

**Pull request merged**